### DISCUSSION TITLE
Nothing, really, nothing works

### DISCUSSION THREAD

**Discussion by ghost** [dani-garcia/vaultwarden#1983] [Q&A]
Oh lord help me though this journey, how can a project have no sensible default, such no-documentation, oh lord.

I tried every-single-tutorial, documentation, this wiki, random indian youtube dudes and mores.
I just wasn't able to get https, that's it. If someone else is reading this feeling sad, it's not you, it's this project.

Let's break it down:
- Use caddy? Oh well you need this special module/plugin thing that we don't have in our docker image
- Want local non https? can't
- Want local https? use a reverse proxy, dns challenge and all the stuff that just end up not working

I'm not a first time linux user, not by far, I tried cloudflare challenge through nginx reverse proxy, caddy, set up A records, install all the things and it doesn't.

The only thing that worked is mobile/desktop app with http but guess what, cannot import without the web app, well f*ck this.

**Comment by Tom41234** [comment]
You need a domain name, then you can setup https. I'm sorry if you already know this, but it's not possible otherwise.
I ended up buying a domain name, setup Nginx Proxy Manager (NPM) and let NPM handle the certificates. Can't get much easier then this.

I did end up setting up a Cloudflare proxy in between, but not before I managed to the certificates running in the manner is described above.

Good luck, it's worth it in the end!

**Comment by BlackDex** [comment]
First, i can understand your frustrations, but minding your language would be appreciated.

Basically, you can just run vaultwarden on your local system accessable via localhost no-https and no settings at all, not even a volume (which would make the whole vault ephemeral, which you probably do not want).

If you want to use it somewhere else besides only localhost you need https, not because we want to, but because the browsers just don't allow access to the crypto-api without it.

Now if you aren't new to Linux you probably have setup some system using a ssl cert either a self-signed, or Lets Encrypt or whatever. That is all what you need to do, make sure there is a ssl cert available. The only caveat here would be that self-signed isn't very well supported on some mobile devices easily.

Also, saying there are no docs makes me think you missed the wiki ( https://github.com/dani-garcia/vaultwarden/wiki ) which explains a lot.

For ssl: https://github.com/dani-garcia/vaultwarden/wiki/Enabling-HTTPS
For reverse proxies: https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples

There are also a lot tutorials on the web, either via a blog or even on YouTube on how to setup Vaultwarden ( or Bitwarden_RS, which is the old name).

If you want more detailed help, it would be best if you can describe which steps you have taken, and what the results are, which errors do you get etc... At this point that is all the information i can give you.

**Comment by NoseyNick** [comment]
For what it's worth, I followed the Apache instructions from the fabulous wiki, on my server that admittedly already had LetsEncrypt SSL certs, but wrapping Apache as an HTTPS proxy around Vaultwarden took... ooooh... maybe 3mins max, and that was only because I didn't have the right proxy modules loaded the first time.

**Comment by redpixy** [comment]
> Oh lord help me though this journey, how can a project have no sensible default, such no-documentation, oh lord.
> 
> I tried every-single-tutorial, documentation, this wiki, random indian youtube dudes and mores.
> I just wasn't able to get https, that's it. If someone else is reading this feeling sad, it's not you, it's this project.
> 
> Let's break it down:
> 
> * Use caddy? Oh well you need this special module/plugin thing that we don't have in our docker image
> * Want local non https? can't
> * Want local https? use a reverse proxy, dns challenge and all the stuff that just end up not working
> 
> I'm not a first time linux user, not by far, I tried cloudflare challenge through nginx reverse proxy, caddy, set up A records, install all the things and it doesn't.
> 
> The only thing that worked is mobile/desktop app with http but guess what, cannot import without the web app, well f*ck this.

There a tutorial for self signed certs. https://github.com/dani-garcia/vaultwarden/wiki/Private-CA-and-self-signed-certs-that-work-with-Chrome

As for your failed challenges from the nginx reverse proxy, it could have been your FW/network settings causing the issues. I have been using nginx reverse proxy for my getting certs and never an issue. The only issue I ever had was due to my router FW. IIRC, nginx do allows you to import your own SSL certs, I never tried that function. 

Did you even check the wiki before ranting here?

**Comment by parker-stephens** [comment]
Using HTTPS is required for most functions of Bitwarden to function on the end-user device, but anywhere in between it is not necessarily a requirement. Point Caddy at your HTTP instance, and it'll do all the magic for you. If it doesn't, that isn't an issue on Vaultwarden's end, but on your end. Try HTTP challenges, try DNS challenges, but **the issue isn't with Vaultwarden**.

**Comment by melbv** [comment]
Oh lord - that is quite a complaint! Everything works fine here and was easy to deploy with a caddy web-server/reverse-proxy in use.
My only issue is that the email function requires a fully-fletched email server (and does not really relay within the localhost networks), that i am reluctant to set up as it will increase my server's attack surface. There are already some funky IPs tracing the smtp connections when i tested the email connection.... The email function works well with external providers though.

**Comment by carceneaux** [comment]
I had zero issues setting this up. I used [this project](https://github.com/dadatuputi/bitwarden_gcloud) which has detailed instructions on how to setup on Google Cloud free tier. It's also mentioned in this project's [wiki](https://github.com/dani-garcia/vaultwarden/wiki/Deployment-examples#google-cloud).
